windows/src/engine/keyman32/calldll.cpp

```diff
@@ -327,46 +588,53 @@ extern "C" BOOL _declspec(dllexport) WINAPI KMDisplayIM(HWND hwnd, BOOL FShowAlw
 	else pt.y += 32; // guessing...
 
 	SetWindowPos(hwnd, HWND_TOPMOST, pt.x,pt.y,0,0, n|SWP_NOSIZE|SWP_SHOWWINDOW|SWP_NOACTIVATE);
-	
+  //SendDebugMessageFormat(0, sdmKeyboard, 0, "KMDisplayIM: Exit");
 	return TRUE;
 }
 
 extern "C" BOOL _declspec(dllexport) WINAPI KMHideIM()
 {
-	if(*Globals::hwndIM() != 0)
+  SendDebugMessageFormat(0, sdmKeyboard, 0, "KMHideIM: Enter");
```

Comment:
Let's comment out these debug messages for now -- no need to log every function call

Reply:
Yes in the commit history I had a commit where I removed a heap of logging. then when I had the emit keystroke issue with the IMX synthesize key up/key down. I had to put it all back in so I didn't want to remove it all again.
It is a shame we don't have levels of logging like debug, warn, error. ( I think info is usually one level too many).  Seeing any code commented out still gives me shivers as it was drummed into me so hard illegal in SIL (Safety Integrity Level) code.  --- doesn't mean I sometimes forget though. I guess I will just remove it.